controllers/k8ssandra/cassandra_telemetry_reconciler.go

```diff
@@ -0,0 +1,94 @@
+// Logic in this file reconciles telemetry resources into the state declared in the CRs.
+
+package k8ssandra
+
+import (
+	"context"
+	"errors"
+	"github.com/go-logr/logr"
+	cassdcapi "github.com/k8ssandra/cass-operator/apis/cassandra/v1beta1"
+	api "github.com/k8ssandra/k8ssandra-operator/apis/k8ssandra/v1alpha1"
+	k8ssandraapi "github.com/k8ssandra/k8ssandra-operator/apis/k8ssandra/v1alpha1"
+	"github.com/k8ssandra/k8ssandra-operator/pkg/result"
+	"github.com/k8ssandra/k8ssandra-operator/pkg/telemetry"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+func (r *K8ssandraClusterReconciler) reconcileCassandraDCTelemetry(
+	ctx context.Context,
+	kc *api.K8ssandraCluster,
+	dcTemplate api.CassandraDatacenterTemplate,
+	actualDc *cassdcapi.CassandraDatacenter,
+	logger logr.Logger,
+	remoteClient client.Client,
+) result.ReconcileResult {
+	logger.Info("reconciling telemetry")
+	mergedSpec := kc.Spec.Cassandra.CassandraTelemetry.Merge(dcTemplate.CassandraTelemetry)
```

Comment:
Bugger! That's a good catch! I am going to make this change immediately actually, because the behaviour for [`TelemetrySpec.Merge()`](https://github.com/Miles-Garnsey/k8ssandra-operator/blob/de687f0219da39a26d537e01273f0ef55f9d5121/apis/telemetry/v1alpha1/telemetry_methods.go#L10) is currently inconsistent with the behaviour for [`PrometheusTelemetrySpec.Merge()`](https://github.com/Miles-Garnsey/k8ssandra-operator/blob/de687f0219da39a26d537e01273f0ef55f9d5121/apis/telemetry/v1alpha1/telemetry_methods.go#L31).

I'll also add logic to bail early if telemetry is `nil`.

Reply:
Actually, I'm removing that early bail when `mergedSpec == nil`. That would prevent deletion of resources if relemetry is removed from the spec. I don't think the optimisation you're proposing is possible.

Reply:
The `Merge()` behaviour changes remain however, that comment was still very valid.